File: lib/features/pages/privacy_screen.dart
import 'package:flutter/material.dart';

import '../../core/design_system/design_system.dart';
import 'page_widgets.dart';

/// Static "Privacy Policy" page, ported faithfully from the web
/// `app/privacy/page.tsx`.
class PrivacyScreen extends StatelessWidget {
  const PrivacyScreen({super.key});

  @override
  Widget build(BuildContext context) {
    return PageScaffold(
      title: 'Privacy Policy',
      children: [
        const SizedBox(height: 4),
        const PageSectionHeading(title: 'Privacy Policy'),
        const SizedBox(height: 24),

        // Introduction.
        const PageContentCard(
          title: 'Introduction',
          icon: Icons.visibility_outlined,
          iconColor: FCPalette.blue600,
          children: [
            PageParagraph(
              'FinnaCalc is committed to protecting your privacy. This Privacy '
              'Policy explains how information is collected, used, disclosed, '
              'and safeguarded when you visit the website and use the financial '
              'calculators and tools.',
            ),
            SizedBox(height: 16),
            PageParagraph(
              'By using FinnaCalc, you agree to the collection and use of '
              'information in accordance with this policy. If you do not agree '
              'with the policies and practices outlined, please do not use the '
              'services.',
            ),
          ],
        ),
        const SizedBox(height: 16),

        // Information Collected.
        PageContentCard(
          title: 'Information Collected',
          icon: Icons.storage_outlined,
          iconColor: FCPalette.green600,
          children: [
            const PageSubheading('Information You Provide'),
            const SizedBox(height: 12),
            const PageBulletList([
              BulletItem(
                'Calculator inputs and financial data (processed locally, not '
                'stored)',
              ),
              BulletItem('Contact information when you reach out'),
              BulletItem('Feedback and suggestions you provide'),
            ]),
            const SizedBox(height: 20),
            const PageSubheading('Automatically Collected Information'),
            const SizedBox(height: 12),
            const PageBulletList([
              BulletItem(
                'Usage data and analytics (page views, time spent, features '
                'used)',
              ),
              BulletItem(
                'Device information (browser type, operating system, screen '
                'resolution)',
              ),
              BulletItem('IP address and general location information'),
              BulletItem('Cookies and similar tracking technologies'),
            ]),
            const SizedBox(height: 16),
            PageNoteBox(
              lead: 'Important:',
              text:
                  'All financial calculations are performed locally in your '
                  'browser. Your personal financial data entered into the '
                  'calculators is not stored, transmitted, or accessed.',
              background: FCPalette.blue50,
              foreground: FCPalette.blue700,
            ),
          ],
        ),
        const SizedBox(height: 16),

        // How Information Is Used.
        const PageContentCard(
          title: 'How Information Is Used',
          icon: Icons.people_outline,
          iconColor: FCPalette.purple600,
          children: [
            PageBulletList([
              BulletItem(
                'To provide and maintain financial calculators and tools',
                lead: 'Service Provision',
              ),
              BulletItem(
                'To analyze usage patterns and improve services',
                lead: 'Improvement',
              ),
              BulletItem(
                'To respond to inquiries and provide customer support',
                lead: 'Communication',
              ),
              BulletItem(
                'To detect, prevent, and address technical issues and security '
                'threats',
                lead: 'Security',
              ),
              BulletItem(
                'To comply with applicable laws and regulations',
                lead: 'Legal Compliance',
              ),
            ]),
          ],
        ),
        const SizedBox(height: 16),

        // Information Sharing and Disclosure.
        const PageContentCard(
          title: 'Information Sharing and Disclosure',
          icon: Icons.lock_outline,
          iconColor: FCPalette.red600,
          children: [
            PageParagraph(
              'Personal information is not sold, traded, or otherwise '
              'transferred to third parties except in the following '
              'circumstances:',
            ),
            SizedBox(height: 16),
            PageBulletList([
              BulletItem(
                'Trusted third parties who assist in operating the website and '
                'conducting business',
                lead: 'Service Providers',
              ),
              BulletItem(
                'When required by law or to protect rights and safety',
                lead: 'Legal Requirements',
              ),
              BulletItem(
                'In connection with a merger, acquisition, or sale of assets',
                lead: 'Business Transfers',
              ),
              BulletItem(
                'When you have given explicit consent for sharing',
                lead: 'Consent',
              ),
            ]),
          ],
        ),
        const SizedBox(height: 16),

        // Data Security.
        PageContentCard(
          title: 'Data Security',
          icon: Icons.shield_outlined,
          iconColor: FCPalette.orange600,
          children: [
            const PageParagraph(
              'Appropriate technical and organizational security measures are '
              'implemented to protect your information against unauthorized '
              'access, alteration, disclosure, or destruction.',
            ),
            const SizedBox(height: 16),
            const PageBulletList([
              BulletItem('SSL encryption for data transmission'),
              BulletItem('Regular security assessments and updates'),
              BulletItem(
                'Limited access to personal information on a need-to-know basis',
              ),
              BulletItem('Secure hosting infrastructure'),
            ]),
            const SizedBox(height: 16),
            PageNoteBox(
              lead: 'Note:',
              text:
                  'While efforts are made to protect your information, no method '
                  'of transmission over the internet or electronic storage is '
                  '100% secure. Absolute security cannot be guaranteed.',
              background: FCPalette.yellow50,
              foreground: FCPalette.yellow800,
            ),
          ],
        ),
        const SizedBox(height: 16),

        // Cookies and Tracking.
        const PageContentCard(
          title: 'Cookies and Tracking Technologies',
          children: [
            PageParagraph(
              'Cookies and similar tracking technologies are used to enhance '
              'your experience on the website:',
            ),
            SizedBox(height: 16),
            PageBulletList([
              BulletItem(
                'Required for basic website functionality',
                lead: 'Essential Cookies',
              ),
              BulletItem(
                'To understand how visitors use the site',
                lead: 'Analytics Cookies',
              ),
              BulletItem(
                'To remember your settings and preferences',
                lead: 'Preference Cookies',
              ),
            ]),
            SizedBox(height: 16),
            PageParagraph(
              'You can control cookies through your browser settings. However, '
              'disabling certain cookies may affect website functionality.',
            ),
          ],
        ),
        const SizedBox(height: 16),

        // Your Privacy Rights.
        const PageContentCard(
          title: 'Your Privacy Rights',
          children: [
            PageParagraph(
              'Depending on your location, you may have the following rights:',
            ),
            SizedBox(height: 16),
            PageBulletList([
              BulletItem(
                'Request information about the personal data held about you',
                lead: 'Access',
              ),
              BulletItem(
                'Request correction of inaccurate or incomplete information',
                lead: 'Correction',
              ),
              BulletItem(
                'Request deletion of your personal information',
                lead: 'Deletion',
              ),
              BulletItem(
                'Request a copy of your data in a structured format',
                lead: 'Portability',
              ),
              BulletItem(
                'Object to certain processing of your information',
                lead: 'Objection',
              ),
            ]),
          ],
        ),
        const SizedBox(height: 16),

        // Children's Privacy.
        const PageContentCard(
          title: "Children's Privacy",
          children: [
            PageParagraph(
              'The services are not intended for children under 13 years of '
              'age. Personal information from children under 13 is not '
              'knowingly collected. If you are a parent or guardian and believe '
              'your child has provided personal information, please make '
              'contact immediately.',
            ),
          ],
        ),
        const SizedBox(height: 16),

        // Changes to This Privacy Policy.
        const PageContentCard(
          title: 'Changes to This Privacy Policy',
          icon: Icons.description_outlined,
          iconColor: FCPalette.blue600,
          children: [
            PageParagraph(
              'This Privacy Policy may be updated from time to time. You will '
              'be notified of any changes by posting the new Privacy Policy on '
              'this page. You are advised to review this Privacy Policy '
              'periodically for any changes.',
            ),
          ],
        ),
        const SizedBox(height: 16),

        // Contact Us.
        const PageContentCard(
          title: 'Contact Us',
          children: [
            PageParagraph(
              'If you have any questions about this Privacy Policy or privacy '
              'practices, please make contact:',
            ),
            SizedBox(height: 16),
            ContactLine(label: 'Help & Assistance', value: 'helpfinnacalc@gmail.com'),
            SizedBox(height: 8),
            ContactLine(label: 'Inquiries', value: 'finnacalc@gmail.com'),
          ],
        ),
      ],
    );
  }
}

File: app/src/main/kotlin/com/finnacalc/android/features/pages/PrivacyScreen.kt
//
// PrivacyScreen.kt
//
// Port of iOS Features/Pages/PrivacyView.swift — the Privacy Policy as a stack
// of titled cards. Copy ported verbatim; rendering is LegalPage.kt.
//

package com.finnacalc.android.features.pages

import android.content.Intent
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Description
import androidx.compose.material.icons.filled.Group
import androidx.compose.material.icons.filled.Lock
import androidx.compose.material.icons.filled.Shield
import androidx.compose.material.icons.filled.Storage
import androidx.compose.material.icons.filled.Visibility
import androidx.compose.runtime.Composable
import androidx.compose.ui.platform.LocalContext

private val privacySections = listOf(
    LegalSection(
        title = "Introduction",
        icon = Icons.Filled.Visibility,
        tint = IconColor.blue,
        blocks = listOf(
            LegalBlock.Paragraph(
                "Your privacy matters to us. This Privacy Policy explains how we collect, use, " +
                    "share, and protect your information when you use the FinnaCalc app, " +
                    "including its calculators, budgeting tools, investing research, guided tax " +
                    "estimator, education content, and the FinnaBot assistant."
            ),
            LegalBlock.Paragraph(
                "This policy covers FinnaCalc everywhere it runs: the apps for phones and tablets " +
                    "and the website at finnacalc.com. By using FinnaCalc, you agree to how we " +
                    "collect and use information under this policy. If you don't agree with it, " +
                    "please don't use the services."
            ),
        ),
    ),
    LegalSection(
        title = "Information Collected",
        icon = Icons.Filled.Storage,
        tint = IconColor.green,
        blocks = listOf(
            LegalBlock.Subheading("Information You Provide"),
            LegalBlock.Bullets(
                listOf(
                    "Optional account details (name, email) if you sign up, stored with our " +
                        "authentication provider (Supabase)",
                    "Calculator, budgeting, and tax-estimator inputs, which are saved on your " +
                        "device, not on our servers",
                    "Bank statements you import, which are parsed and stored only on your device",
                    "Messages you send to FinnaBot and the Budget Analysis assistant",
                    "Contact information and feedback when you reach out",
                )
            ),
            LegalBlock.Subheading("Automatically Collected Information"),
            LegalBlock.Bullets(
                listOf(
                    "The stock symbols and market pages you request (needed to fetch quotes, " +
                        "charts, and news)",
                    "Basic server logs (IP address, timestamps) when the app or website talks to " +
                        "our services",
                    "Device information such as OS version",
                )
            ),
            LegalBlock.Subheading("Bank & Brokerage Connections"),
            LegalBlock.Bullets(
                listOf(
                    "Bank and brokerage links run through Plaid and SnapTrade. Your bank " +
                        "credentials go to them directly and never touch FinnaCalc's servers",
                    "Imported transactions and holdings are stored on your device",
                )
            ),
            LegalBlock.Callout(
                lead = "Important:",
                body = " Your budget, goals, history, and tax-estimator answers live on your " +
                    "device. Sensitive tax fields (Social Security numbers, bank details) are " +
                    "never saved, and the tax estimator does not transmit your return, since " +
                    "e-filing isn't enabled yet.",
                tone = CalloutTone.Info,
            ),
        ),
    ),
    LegalSection(
        title = "How Information Is Used",
        icon = Icons.Filled.Group,
        tint = IconColor.purple,
        blocks = listOf(
            LegalBlock.LeadBullets(
                listOf(
                    "Service Provision:" to
                        " To provide calculators, budgeting, market data, tax estimation, and the " +
                        "AI assistant",
                    "Personalization:" to
                        " To answer FinnaBot and Budget Analysis questions using the budget " +
                        "snapshot you share in that conversation",
                    "Improvement:" to " To understand usage and improve the apps and website",
                    "Communication:" to " To respond to inquiries and provide support",
                    "Security:" to
                        " To detect, prevent, and address technical issues and security threats",
                    "Legal Compliance:" to " To comply with applicable laws and regulations",
                )
            )
        ),
    ),
    LegalSection(
        title = "Information Sharing and Disclosure",
        icon = Icons.Filled.Lock,
        tint = IconColor.red,
        blocks = listOf(
            LegalBlock.Paragraph(
                "Personal information is not sold, traded, or otherwise transferred to third " +
                    "parties except in the following circumstances:"
            ),
            LegalBlock.LeadBullets(
                listOf(
                    "Service Providers:" to
                        " Supabase (accounts), Plaid and SnapTrade (bank/brokerage links, under " +
                        "their own privacy policies), Google (AI responses for FinnaBot), and " +
                        "market-data providers that receive the ticker symbols you view",
                    "Purchases:" to
                        " Payment is handled by whichever processor you subscribe through: the " +
                        "app store you bought from (such as the Apple App Store or Google Play) " +
                        "or our payment processor on the website. They hold your card details; " +
                        "FinnaCalc only learns which plan is active",
                    "Legal Requirements:" to
                        " When required by law or to protect rights and safety",
                    "Business Transfers:" to
                        " In connection with a merger, acquisition, or sale of assets",
                    "Consent:" to " When you have given explicit consent for sharing",
                )
            ),
        ),
    ),
    LegalSection(
        title = "Data Security",
        icon = Icons.Filled.Shield,
        tint = IconColor.orange,
        blocks = listOf(
            LegalBlock.Paragraph(
                "We use sensible technical and organizational safeguards to protect your " +
                    "information from unauthorized access, alteration, disclosure, or loss."
            ),
            LegalBlock.Bullets(
                listOf(
                    "Personal finance data (budget, goals, tax answers) stays on your device",
                    "SSL/TLS encryption for everything sent to our services",
                    "Bank credentials handled only by Plaid and SnapTrade, never by FinnaCalc",
                    "Limited access to personal information on a need-to-know basis",
                    "Secure hosting infrastructure",
                )
            ),
            LegalBlock.Callout(
                lead = "Note:",
                body = " While efforts are made to protect your information, no method of " +
                    "transmission over the internet or electronic storage is 100% secure. " +
                    "Absolute security cannot be guaranteed.",
                tone = CalloutTone.Caution,
            ),
        ),
    ),
    LegalSection(
        title = "On-Device Storage & Preferences",
        blocks = listOf(
            LegalBlock.Paragraph(
                "The app stores your working data and preferences locally on your device rather " +
                    "than with tracking cookies:"
            ),
            LegalBlock.LeadBullets(
                listOf(
                    "Your data:" to
                        " Budget items, savings and investing goals, history snapshots, " +
                        "watchlist, and tax-estimator answers",
                    "Preferences:" to
                        " Appearance (light/dark), chart settings, and similar choices",
                    "Notifications:" to
                        " Goal alerts and bill reminders are scheduled on your device; nothing " +
                        "about them is sent to a server",
                )
            ),
            LegalBlock.Paragraph(
                "Deleting the app removes this local data (the website keeps its own in your " +
                    "browser). You can also clear budgeting data from the Budgeting page and " +
                    "restart the tax estimator at any time. FinnaCalc does not use advertising " +
                    "trackers."
            ),
        ),
    ),
    LegalSection(
        title = "Your Privacy Rights",
        blocks = listOf(
            LegalBlock.Paragraph("Depending on your location, you may have the following rights:"),
            LegalBlock.LeadBullets(
                listOf(
                    "Access:" to " Request information about the personal data held about you",
                    "Correction:" to " Request correction of inaccurate or incomplete information",
                    "Deletion:" to " Request deletion of your personal information",
                    "Portability:" to " Request a copy of your data in a structured format",
                    "Objection:" to " Object to certain processing of your information",
                )
            ),
        ),
    ),
    LegalSection(
        title = "Children's Privacy",
        blocks = listOf(
            LegalBlock.Paragraph(
                "The services are not intended for children under 13 years of age. Personal " +
                    "information from children under 13 is not knowingly collected. If you are a " +
                    "parent or guardian and believe your child has provided personal information, " +
                    "please make contact immediately."
            )
        ),
    ),
    LegalSection(
        title = "Changes to This Privacy Policy",
        icon = Icons.Filled.Description,
        tint = IconColor.blue,
        blocks = listOf(
            LegalBlock.Paragraph(
                "This Privacy Policy may be updated from time to time. You will be notified of " +
                    "any changes by posting the new Privacy Policy on this page. You are advised " +
                    "to review this Privacy Policy periodically for any changes."
            )
        ),
    ),
    LegalSection(
        title = "Contact Us",
        blocks = listOf(
            LegalBlock.Paragraph(
                "If you have any questions about this Privacy Policy or privacy practices, " +
                    "please make contact:"
            ),
            LegalBlock.Contact("Help:", "helpfinnacalc@gmail.com"),
            LegalBlock.Contact("Inquiries:", "finnacalc@gmail.com"),
        ),
    ),
)

@Composable
fun PrivacyScreen() {
    val context = LocalContext.current
    LegalPage("Privacy Policy", effectiveDate = null, sections = privacySections) { email ->
        context.startActivity(Intent(Intent.ACTION_VIEW, android.net.Uri.parse("mailto:$email")))
    }
}
